### PR TITLE
Add Pay Now feature on Adobe-commerce

### PR DESCRIPTION
**Pull request by Francois-Gomis**
### Reason for change

[Linear task](https://linear.app/almapay/issue/MPP-84/deploy-pay-now)

### Code changes
Add pay now in back office - checkout and widget

### How to test

Add test API key and check fee plans
Check product page and checkout 

### Checklist for authors and reviewers

- [x] The title of the PR uses business wording, not technical jargon, for the changelog readers to understand it
- [x] The PR implements the changes asked in the referenced task / issue
- [x] The automated tests are compliant with the [testing strategy](https://www.notion.so/almapay/Backend-testing-strategy-06c642cec1bf47b9b8feca3a91ea8d4a)
- [x] The tests are relevant, and cover the corner/error cases, not only the happy path
- [x] You understand the impact of this PR on existing code/features
- [x] The changes include adequate logging and Datadog traces
- [x] Documentation is updated (API, developer documentation, ADR, Notion...)

### CODE REVIEW

File: Helpers/PaymentPlansHelper.php [hunk 1]
```diff
@@ -153,9 +159,16 @@ public function planLabelByKey(string $key): string
         if (isset($matches[1])) {
             $label =  __('Pay in %1 installments', $matches[1]);
         }
+        // TODO refactor with basket items PR
+        if (isset($matches[1]) && $matches[1] === '1'  & $matches[2] == 0 && $matches[3] == 0) {
```
Comment (Benjamin-Freoua-Alma) [inline Helpers/PaymentPlansHelper.php:163]: matches[1] is in `string` ? and matches[2] and [3] in `int` ?

**Review by Benjamin-Freoua-Alma — APPROVED** [review]

File: CustomerData/AlmaSection.php [hunk 2]
```diff
@@ -27,11 +27,14 @@ class AlmaSection implements SectionSourceInterface
     public function __construct(
         Eligibility $eligibility,
         CheckoutConfigHelper $checkoutConfigHelper
-    )
-    {
+    ) {
```
Comment (ClaireAlma) [inline CustomerData/AlmaSection.php:30]: manque commentaire de fonction

File: Helpers/PaymentPlansHelper.php (hunk 1 above)
Comment (ClaireAlma) [inline Helpers/PaymentPlansHelper.php:163]: tu as oublié le isset de 2 et de 3 ici :)

**Review by ClaireAlma — CHANGES_REQUESTED** [review]

File: Helpers/PaymentPlansHelper.php (hunk 1 above)
Comment (joyet-simon) [reply Helpers/PaymentPlansHelper.php:163]: `===` make a verification on the value and the type. `==` make a verification only on the value.

**Review by joyet-simon — APPROVED** [review]

**Review by joyet-simon — APPROVED** [review]

File: Helpers/PaymentPlansHelper.php (hunk 1 above)
Comment (Francois-Gomis) [reply Helpers/PaymentPlansHelper.php:163]: Done

**Review by ClaireAlma — APPROVED** [review]

**Pull request merged**